### PR TITLE
Cleanups and optimizations for composite ops, tile cmd lists.

### PR DESCRIPTION
**Pull request by glennw**
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/416)

### CODE REVIEW

**Comment by glennw** [comment]
r? @pcwalton 

These are a small performance optimization, but the main point is to remove the simple cmd list code, so there is only one path for adding primitives (this will be useful once I start making use of prim caching).

File: webrender/src/tiling.rs [hunk 1]
```diff
@@ -2681,7 +2625,7 @@ impl FrameBuilder {
                       transform: Matrix4D<f32>,
                       pipeline_id: PipelineId,
                       scroll_layer_id: ScrollLayerId,
-                      composition_operations: Vec<CompositionOp>) {
+                      composition_operations: &Vec<CompositionOp>) {
```
Comment (pcwalton) [inline webrender/src/tiling.rs]: nit: could just be `composition_operations: &[CompositionOp]`

**Comment by pcwalton** [comment]
r=me with that nit if you like

**Pull request merged**